api-tests/core/content-manager/find-relations.test.api.js

```diff
@@ -0,0 +1,803 @@
+            test(`when entity ID is ${
+              !useEmptyShop ? 'undefined' : 'an empty entity'
+            }`, async () => {
+              const { modelUID, idEmptyShop } = isComponent
+                ? data.testData.component
+                : data.testData.entity;
+              const id = useEmptyShop ? idEmptyShop : undefined;
+
+              const res = await rq({
+                method: 'GET',
+                url: `/content-manager/relations/${modelUID}/${fieldName}`,
+                qs: {
+                  ...qs,
+                  id,
+                },
+              });
+              expect(res.status).toBe(200);
+
+              const productsInThisLocale = data.products[status].filter(
+                // This test is running for the default locale (en)
+                // any products that are in the non default locale should not
+                // be considered available
+                (product) => product.locale === defaultLocale
+              );
+
+              expect(res.body.results.map((result) => result.id)).toMatchObject(
+                productsInThisLocale
+                  // Results form the request should be sorted by name
+                  // but are not necessarily in the same order as data.products
+                  .sort((a, b) => a.name.localeCompare(b.name))
+                  .map((product) => product.id)
+              );
+
+              const idsToOmit = [productsInThisLocale[1]?.id].filter(Boolean);
+              const omitIdsRes = await rq({
+                method: 'GET',
+                url: `/content-manager/relations/${modelUID}/${fieldName}`,
+                qs: {
+                  ...qs,
+                  id,
+                  idsToOmit,
+                },
+              });
+
+              expect(omitIdsRes.body.results).toHaveLength(
+                productsInThisLocale
+                  .map((product) => product.id)
+                  .filter((id) => !idsToOmit.includes(id)).length
+              );
+            });
+          }
+        );
+
+        test(`Get relations for ${fieldName}`, async () => {
+          const { id, modelUID } = isComponent ? data.testData.component : data.testData.entity;
+
+          const res = await rq({
+            method: 'GET',
+            url: `/content-manager/relations/${modelUID}/${fieldName}`,
+            qs: {
+              ...qs,
+              id,
+            },
+          });
+          expect(res.status).toBe(200);
+
+          // Get the ids of the products that are not already related to this
+          // entity and are in the same locale as the entity
+          const availableProducts = data.products[status]
+            .filter((product) => {
+              return (
+                !getRelatedProductIds(isComponent, status, fieldName).includes(product.id) &&
+                product.locale === defaultLocale
+              );
+            })
+            .sort((a, b) => a.name.localeCompare(b.name))
```

Comment:
why do we sort by name? shouldn't we let the database sort by the relation order?

Reply:
( but don't know if we provide a feature to sort by the main field )

Reply:
Here I'm filtering `data.products[status]` to find the available products. Which
is created like 
```
    data.products.draft.push(
      skate.data,
      chair.data,
      candle.data,
      table.data,
      porte.data,
      fenetre.data
    );
```
FindAvailable should sort by the mainField but the order of
`data.products[status]` is not necessarily right (e.g. alphabetical) so I sort
it to build the expected response. Do you think that's okay?

Reply:
I would say it should, this will just be a select drop down so no reordering available and I think sorting my the mainField will make it easy to look through in most situations. Its how we do it in the current version of findAvailable and there's also search on this endpoint